pkg/sdk/sdk_test.go

```diff
@@ -0,0 +1,390 @@
+				svr.Close()
+			},
+		},
+		{
+			name:     "Failure:: Replace :: incorrect file type", //doubt
+			filePath: "./../../Dockerfile",
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/register{id}", http.MethodPut, func(w http.ResponseWriter, r *http.Request) {
+					err := r.ParseMultipartForm(10000) //File size to come from config
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrFileSizeExceeded), nil)
+						log.Error(err.Error())
+						return
+					}
+					file, _, err := r.FormFile("file")
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrFileParseFail), nil)
+						log.Error(err.Error())
+						return
+					}
+					defer file.Close()
+
+					response.ToJson(w, http.StatusCreated, "SUCCESS", map[string]interface{}{
+						"id": gomock.Any(),
+					})
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err.Error() != "non success status code received : 404" {
+					t.Errorf("Want: %v, Got: %v", "non success status code received : 404", err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			svr := tt.setupFunc()
+			defer tt.cleanupFunc(svr)
+
+			calls := NewHtmlToPdfSvc(svr.URL)
+			fileBytes, err := os.ReadFile(tt.filePath)
+
+			err = calls.Replace(fileBytes, tt.id)
+
+			tt.ValidateFunc(err)
+		})
+	}
+}
+func Test_GeneratePdf(t *testing.T) {
+	tests := []struct {
+		name              string
+		id                string
+		data              map[string]interface{}
+		setupFunc         func() *httptest.Server
+		mockServerHandler func(w http.ResponseWriter, r *http.Request)
+		ValidateFunc      func(err error)
+		cleanupFunc       func(*httptest.Server)
+		expectedResponse  model.Response
+	}{
+		{
+			name: "Success:: GeneratePdf",
+			id:   "1",
+			data: map[string]interface{}{"id": "1"},
+			setupFunc: func() *httptest.Server {
+				svr := testServer("/v1/generate/{id}", http.MethodPost, func(w http.ResponseWriter, r *http.Request) {
+					vars := mux.Vars(r)
+					//we take id as a parameter from url path
+					id, ok := vars["id"]
+					if !ok {
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrIdNeeded), nil)
+						return
+					}
+					var data modelV.GenerateReq
+					err := json.NewDecoder(r.Body).Decode(&data)
+					if err != nil {
+						response.ToJson(w, http.StatusBadRequest, codes.GetErr(codes.ErrDecodingData), nil)
+						log.Error(err.Error())
+						return
+					}
+					data.Id = id
+
+					response.ToJson(w, http.StatusOK, "", nil)
+				})
+				return svr
+			},
+			ValidateFunc: func(err error) {
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err.Error())
+				}
+			},
+			cleanupFunc: func(svr *httptest.Server) {
+				svr.Close()
+			},
+		},
+		{
+			name: "Failure:: GeneratePdf :: id required",
```

Comment:
Which scenario of the sdk function flow is being tested here? ID required is a test case that is handled at server side and since the server is mocked, this can be neglected.

Reply:
removed it
Done